File: app/src/main/java/se/pederjonsson/apps/quizkids/components/NavbarView.java
package se.pederjonsson.apps.quizkids.components;

import android.content.Context;
import android.graphics.Color;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.makeramen.roundedimageview.RoundedImageView;

import java.util.List;

import butterknife.BindView;
import butterknife.BindViews;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import se.pederjonsson.apps.quizkids.Objects.CategoryItem;
import se.pederjonsson.apps.quizkids.Objects.Profile;
import se.pederjonsson.apps.quizkids.R;

/**
 * Created by Gaming on 2018-04-01.
 */

public class NavbarView extends RelativeLayout {

    @BindView(R.id.title)
    TextView title;

    @BindView(R.id.rootview)
    RelativeLayout rootView;

    @BindView(R.id.dotscontainer)
    LinearLayout dotsContainer;

    @BindViews({R.id.dot1, R.id.dot2, R.id.dot3, R.id.dot4, R.id.dot5, R.id.dot6, R.id.dot7, R.id.dot8, R.id.dot9, R.id.dot10})
    List<RoundedImageView> dots;

    private CategoryItem categoryItem;
    private Context mContext;
    private Unbinder unbinder;
    private Profile playingProfile;

    public NavbarView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }



    protected void init(Context context) {
        mContext = context;
        inflate();
    }

    protected void inflate() {
        inflate(mContext, R.layout.navbar, this);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        unbinder = ButterKnife.bind(this);
    }

    public void setUp() {
    }

    public void showTitle(String titlestring){
        title.setText(titlestring);
    }

    public void show(boolean visible){
        if(visible){
            rootView.setVisibility(VISIBLE);
        } else {
            rootView.setVisibility(GONE);
        }
    }

    public void showScore(){
        dotsContainer.setVisibility(VISIBLE);
    }

    public void setScore(boolean correct, int question){
        if(correct){
            dots.get(question).setImageDrawable(mContext.getDrawable(R.drawable.greenrectangle));
        } else {
            dots.get(question).setImageDrawable(mContext.getDrawable(R.drawable.redrectangle));
        }
    }

    public void clearScore(){
        for (RoundedImageView dot: dots){
            dot.setImageDrawable(mContext.getDrawable(R.drawable.whiterectangle));
        }
    }

}

File: app/src/main/java/se/pederjonsson/apps/quizkids/viewcomponents/NavbarView.kt
package se.pederjonsson.apps.quizkids.viewcomponents

import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import com.makeramen.roundedimageview.RoundedImageView
import kotlinx.android.synthetic.main.navbar.view.*
import se.pederjonsson.apps.quizkids.R

/**
 * Created by Gaming on 2018-04-01.
 */
class NavbarView(context: Context?, attrs: AttributeSet?) : RelativeLayout(context, attrs) {

    private var dots:List<RoundedImageView>? = null
    private var mContext: Context? = null
    private var scoreVisible = false

    init{
        mContext = context
        inflate()

    }

    private fun inflate() {
        inflate(mContext, R.layout.navbar, this)
        dots = listOf(dot1, dot2, dot3, dot4, dot5, dot6, dot7, dot8, dot9, dot10)
    }

    fun setUp() {}
    fun showTitle(titlestring: String?) {
        title.text = titlestring;
    }

    fun show(visible: Boolean) {
        if (visible) {
            navbar_rootview.visibility = VISIBLE;
        } else {
            navbar_rootview.visibility = GONE;
        }
    }

    fun showScore(_val: Boolean) {
        scoreVisible = _val
        if (_val) {
            dotscontainer.visibility = VISIBLE;
        } else {
            dotscontainer.visibility = INVISIBLE;
        }
    }

    fun setScore(correct: Boolean, question: Int) {
        dots?.let {
            if (scoreVisible) {
                if (correct) {
                    it[question].setImageDrawable(mContext?.getDrawable(R.drawable.greenrectangle));
                } else {
                    it[question].setImageDrawable(mContext?.getDrawable(R.drawable.redrectangle));
                }
            }
        }
    }

    fun clearScore() {
        dots?.let {
            for (dot in it) {
                dot?.setImageDrawable(mContext?.getDrawable(R.drawable.whiterectangle))
            }
        }
    }

}